dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/ownership/TrackerOwnershipController.java

```diff
@@ -97,15 +98,16 @@ public WebMessage updateTrackerProgramOwner(
             "trackedEntityInstance", trackedEntityInstance, "trackedEntity", trackedEntity);
 
     trackerOwnershipAccessManager.transferOwnership(
-        trackedEntityService.getTrackedEntity(trackedEntityUid),
+        trackedEntityService.getTrackedEntity(
+            trackedEntityUid, UID.of(program), TrackedEntityParams.TRUE),
```

Comment:
Do you need to get enrollments, ... as defined in `TrackedEntityParams.TRUE`? Double-check to only fetch what's really needed.